### PR TITLE
Add Makefile and move integration tests into their own directory

### PR DESCRIPTION
**Pull request by ptrimble**

### CODE REVIEW

**Comment by ptrimble** [comment]
I believe this is ready. As an explanation, I am a fan of simple `make` targets. My goal for this project is to split out the definitions and write libraries for other languages. In this potential future I would love for users to be able to run tests (and other 'common' actions) via the same command, i.e. `make test` in Go and `make test` in Swift and `make test` in Ruby.

@ttwo32 Can you look this over? No breaking change, just behind-the-scenes stuff that will only be interesting to you and me, I think. 😄

**Comment by ptrimble** [comment]
Also, I don't know why the coverage isn't reporting. I'll give it some time (Travis seems to be slow today) and if it still doesn't get reported I'll resubmit the test.

**Comment by ttwo32** [comment]
@ptrimble I have looked this over and tested all make commands.
Those are good!!

File: test/test_helper.rb [hunk 1]
```diff
@@ -1,6 +1,3 @@
-require 'coveralls'
-Coveralls.wear!
-
 $:.unshift(File.expand_path(File.dirname(__FILE__) + '../../lib/'))
```
Comment (ttwo32) [inline test/test_helper.rb]: Maybe, not send coverage occurred as a result of deleting lines??

File: test/integration/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Integration tests
+
+These tests are dependent on the files in /definitions (and, by proxy, /lib/generated_definitions).
+It is possible that these tests will break because of 'unrelated' definition changes. The code
+behind these changes could still be good but since the defintions changed we could see failures.
+
+These are not unit tests. This is not testing whether specific internal code is working. These are
+tests from the consumer perspective. You must recognize that this could fail because of code
+changes unrelated ot definition changes.
```
Comment (ptrimble) [inline test/integration/README.md]: Typo here!

File: test/test_helper.rb (hunk 1 above)
Comment (ptrimble) [inline test/test_helper.rb]: Maybe, but I thought I moved these. Did I forget to put it back?

Comment (ptrimble) [inline test/test_helper.rb]: I think you are right! I don't know why I took these out, to be perfectly honest.

Comment (ptrimble) [inline test/test_helper.rb]: Addressed in rebase.

File: test/integration/README.md (hunk 2 above)
Comment (ptrimble) [inline test/integration/README.md]: Addressed in rebase.

File: test/test_helper.rb (hunk 1 above)
Comment (ptrimble) [inline test/test_helper.rb]: Oh, I remember now! It's because `.simplecov` has the 'require' commands as well. I'm going to try to simplify this....

**Comment by ptrimble** [comment]
There we go! Merging...

**Pull request merged**